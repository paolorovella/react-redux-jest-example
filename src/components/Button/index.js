import styled from 'styled-components';
import theme from 'src/config/theme';

const Button = styled.button`
  color: ${props => (
    theme.buttons[props.type].default ||
    theme.buttons.primary.default
  )};
  border: 2px solid ${props => (
    theme.buttons[props.type].default ||
    theme.buttons.primary.default
  )};
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;

  &:hover {
    background-color: ${props => (
      theme.buttons[props.type].hover ||
      theme.buttons.primary.hover
    )};
    color: white;
    border: 2px solid ${props => (
      theme.buttons[props.type].hover ||
      theme.buttons.primary.hover
    )};
  }
`;

Button.defaultProps = {
  type: 'primary',
};

export default Button;
