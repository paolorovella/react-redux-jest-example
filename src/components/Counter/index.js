import React from 'react';
import PropTypes from 'prop-types';

const Counter = ({ label, value }) => <span>{label}: {value}</span>;

Counter.defaultProps = {
  label: 'Counter',
  value: 0,
};

Counter.propTypes = {
  label: PropTypes.string,
  value: PropTypes.number,
};

export default Counter;
