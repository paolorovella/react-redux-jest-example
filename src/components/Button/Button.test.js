import React from 'react';
import renderer from 'react-test-renderer';
import theme from 'src/config/theme';
import 'jest-styled-components';

import { Button } from 'src/components';

it('<Button /> (styled-component) renders without crashing', () => {
  const tree = renderer.create(
    <Button />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Button /> (styled-component) renders with type prop', () => {
  const tree = renderer.create(
    <Button type="success" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Button /> (styled-component) renders with type prop and match style rules', () => {
  const tree = renderer.create(
    <Button type="success" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', theme.buttons.success.default);
  expect(tree).toHaveStyleRule('border', `2px solid ${theme.buttons.success.default}`);
});
