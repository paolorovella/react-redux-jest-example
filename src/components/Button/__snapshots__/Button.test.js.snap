// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> (styled-component) renders with type prop 1`] = `
.c0 {
  color: #2E7D32;
  border: 2px solid #2E7D32;
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;
}

.c0:hover {
  background-color: #1B5E20;
  color: white;
  border: 2px solid #1B5E20;
}

<button
  className="c0"
  label="Button"
  type="success"
>
  Button
</button>
`;

exports[`<Button /> (styled-component) renders with type prop and match style rules 1`] = `
.c0 {
  color: #2E7D32;
  border: 2px solid #2E7D32;
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;
}

.c0:hover {
  background-color: #1B5E20;
  color: white;
  border: 2px solid #1B5E20;
}

<button
  className="c0"
  label="Button"
  type="success"
>
  Button
</button>
`;

exports[`<Button /> (styled-component) renders without crashing 1`] = `
.c0 {
  color: #1565C0;
  border: 2px solid #1565C0;
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;
}

.c0:hover {
  background-color: #0D47A1;
  color: white;
  border: 2px solid #0D47A1;
}

<button
  className="c0"
  label="Button"
  type="primary"
>
  Button
</button>
`;
